test/mainnet/uniswapV3Router/uniswapV3Router.test.ts

```diff
@@ -0,0 +1,117 @@
+import hre from "hardhat";
+import axios from "axios";
+import { expect } from "chai";
+const { ethers } = hre; //check
+import { BigNumber } from "bignumber.js";
+import { deployAndEnableConnector } from "../../../scripts/tests/deployAndEnableConnector";
+import { buildDSAv2 } from "../../../scripts/tests/buildDSAv2";
+import { encodeSpells } from "../../../scripts/tests/encodeSpells";
+import { getMasterSigner } from "../../../scripts/tests/getMasterSigner";
+import { addresses } from "../../../scripts/tests/mainnet/addresses";
+import { addLiquidity } from "../../../scripts/tests/addLiquidity";
+import { abis } from "../../../scripts/constant/abis";
+import { ConnectV2UniswapV3AutoRouter__factory } from "../../../typechain";
+import er20abi from "../../../scripts/constant/abi/basics/erc20.json";
+import type { Signer, Contract } from "ethers";
+import { CurrencyAmount, Token, TradeType, Currency, Percent } from "@uniswap/sdk-core";
+import { AlphaRouter } from "@uniswap/smart-order-router";
+const provider = new ethers.providers.JsonRpcProvider("https://mainnet.infura.io/v3/47722f74a0264bd98bdc66c7f0177333");
+const router = new AlphaRouter({ chainId: 1, provider });
```

Comment:
```suggestion
const router = new AlphaRouter({ chainId: 1, provider: ethers.provider});
```

@bhavik-m Can you try like this?